modules/ROOT/pages/java/java-create-instance.adoc

```diff
@@ -8,27 +8,71 @@ endif::[]
 
 
 
-This example shows how to create an instance of a class:
+This example shows how to create an instance of a class whose constructor does not require parameters:
 
 [source,xml,linenums]
 ----
 <java:new class="com.foo.Person" constructor="Person()"/>
 ----
 
-This example shows how to create an instance using the default constructor. You can also use a non-custom constructor.
+This example shows how to create an instance of a class when the constructor requires parameters:
+
 [source,xml,linenums]
 ----
 <java:new class="com.foo.Person" constructor="Person(String, Integer)">
     <java:args>#[{
-      firstName: 'MG',
-      age: 34
+      name: 'Jhon Doe',
+      age: 30
     }]</java:args>
 </java:new>
 ----
 
-IMPORTANT: When configuring the constructor arguments in the `args` parameter, the keys of the map will identify which parameter should be invoked. *In order to reference the parameters by name* (`firstName`, `age`, etc.), the java class containing the Method or Constructor has to be compiled using https://docs.oracle.com/javase/tutorial/reflect/member/methodparameterreflection.html[the `-parameters` compiler flag]. If the class was not compiled with this flag, then the same parameters should be refenced in the canonical way as `arg0`, `arg1`, etc.
+IMPORTANT: When configuring the constructor arguments in the `<java:args>` parameter,
+the keys of the map will determine how the parameters will be passed to the constructor. *In
+order to reference the parameters by name* (`firstName`, `age`, etc.),
+the java class containing the Method or Constructor has to be compiled
+using https://docs.oracle.com/javase/tutorial/reflect/member/methodparameterreflection.html[the `-parameters` compiler flag].
+If the class was not compiled with this flag, then the same parameters
+should be referenced in the declared order and with the canonical names (`arg0: 'Jhon Doe'`, `arg1: 30`, etc.).
+
+[TIP]
+===============================
+If your Java classes are defined in your Studio Project, you will need to add
+the Maven Compiler plugin as a dependency in you `pom.xml` in order to compile your classes
+with the `-parameters` flag:
+[source,xml,linenums]
+----
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-compiler-plugin</artifactId>
+    <version>3.8.1</version>
+    <configuration>
+        <compilerArgs>
+            <arg>-parameters</arg>
+        </compilerArgs>
+    </configuration>
+</plugin>
+----
+===============================
+
+One of the advantages of using the Java Module is that you will get https://docs.mulesoft.com/studio/7.3/datasense-concept[DataSense] support,
+which can help you build the `<java:args>` map and will also provide Metadata for the
+operation's output. In the above examples, the `Person` object will be placed in the
```

Comment:
Add an image of how people are going to see this Metadata benefits on Studio

Reply:
added